elastic/elastic-agent
# [helm] remove unsupported Release fields for publishing the chart as EKS addon

## What does this PR do?

This PR removes unsupported fields `Release.IsInstall`, `Release.IsUpgrade`, and `.Release.Service` from the Helm chart. These fields are unavailable in the EKS addon and their presence leads to Helm chart validation failures when using the AWS console. The changes ensure compatibility with EKS addon requirements by refactoring the chart to exclude these fields.

## Why is it important?

This change is crucial for ensuring that the Helm chart used in the Elastic Agent EKS addon passes AWS console validations. Unsupported fields prevent us from publishing this Helm chart as an EKS addon.

## Checklist


## Disruptive User Impact

This PR should not cause any direct disruption to users, but it will ensure smoother integration and deployment in AWS EKS environments.

## How to test this PR locally

1. Try to publish the packaged Helm chart as an EKS add-on and you will see no validation errors 🙂 

## Related issues

- Closes https://github.com/elastic/elastic-agent/issues/6053
